### PR TITLE
User transactions

### PR DESCRIPTION
**Pull request by josenavas**
Built on top of https://github.com/biocore/qiita/pull/1291

Tests are expected to fail. I made a change in one util function that broke the study object. I will fix the study object in the next PR, because I don't want to make this PR excessively big.

### CODE REVIEW

**Comment by squirrelo** [comment]
:+1:  if tests pass

File: qiita_db/user.py [hunk 1]
```diff
@@ -126,48 +126,56 @@ def login(cls, email, password):
...
+            # pull password out of database
+            sql = ("SELECT password, user_level_id FROM qiita.{0} WHERE "
+                   "email = %s".format(cls._table))
+            trans.add(sql, [email])
+
+            info = trans.execute()[-1][0]
+
+            # verify user email verification
+            # MAGIC NUMBER 5 = unverified email
+            if int(info[1]) == 5:
```
Comment (antgonza) [inline qiita_db/user.py]: I know is not part of your code but can we use the function that retrieves the id from the value?

**Comment by josenavas** [comment]
Thanks @antgonza 
Ready for review

**Comment by antgonza** [comment]
Real errors ...

**Comment by josenavas** [comment]
Yes, they're expected, check the comment of the PR.

**Comment by antgonza** [comment]
Ohh! :smile:

**Pull request merged**